tests/test_cli.py

```diff
@@ -75,3 +75,11 @@ def test_no_traces(store_data, stdout, stderr):
     assert stderr.getvalue() == "No traces found\n"
     assert stdout.getvalue() == ''
     assert ret == 0
+
+
+def test_cli_context_manager_activated(stdout, stderr):
+    with mock.patch.object(DefaultConfig, 'cli_context') as mock_context:
```

Comment:
Instead of patching, let's define a real config with a toy `cli_context` that is an actual context manager that does something (doesn't really matter what, anything you can assert on; append something to a global list, print something to stdout, whatever you like), then assert on the thing the context manager does. This is closer to the real usage pattern and gives better confidence things are working right; patching in tests should be just a fallback for when the alternative is way too painful.